syntax = "proto2";

import "api/player_id.proto";

package api.metadata;

message GameScore {
    optional int32 team0score = 1;
    optional int32 team1score = 2;
}

message Team {
    repeated api.PlayerId player_ids = 1; // Player.id
    optional int32 score = 2;
    optional bool is_orange = 3;
}


message Goal {
    optional int32 frame = 1;
    optional api.PlayerId player_id = 2; // Player.id
}

message Demo {
    optional int32 frame = 1;
    optional api.PlayerId attacker_id = 2; // Player.id
    optional api.PlayerId victim_id = 3; // Player.id
}

message GameMetadata {
    optional int32 id = 1;
    optional string name = 2;
    optional string map = 3;
    optional int32 version = 4;
    optional int64 time = 5;
    optional int32 frames = 6;
    optional GameScore score = 7;
    optional Team teams = 8;
    optional Goal goals = 9;
    optional Demo demos = 10;
}
